services/src/main/java/org/exoplatform/gamification/storage/dao/GamificationDAO.java

```diff
@@ -0,0 +1,113 @@
+package org.exoplatform.gamification.storage.dao;
+
+import org.exoplatform.commons.persistence.impl.GenericDAOJPAImpl;
+import org.exoplatform.gamification.entities.domain.effective.GamificationContextEntity;
+import org.exoplatform.gamification.service.effective.GamificationSearch;
+import org.exoplatform.gamification.service.effective.Leaderboard;
+import org.exoplatform.gamification.service.effective.Piechart;
+
+import javax.persistence.NoResultException;
+import javax.persistence.PersistenceException;
+import javax.persistence.TypedQuery;
+import java.util.List;
+
+public class GamificationDAO extends GenericDAOJPAImpl<GamificationContextEntity, Long> {
```

Comment:
DAO classes should be interfaces, and then a JPA implementation should be created. Then only the interface must be used in services.